# JS Error

**Try** deyimi, hatalar iÃ§in bir kod bloÄŸunu test etmenizi saÄŸlar.

**Catch** deyimi, hatayÄ± iÅŸlemenizi saÄŸlar.

**Throw** ifadesi, Ã¶zel hatalar oluÅŸturmanÄ±za olanak tanÄ±r.

**Finally** deyimi, sonuca bakÄ±lmaksÄ±zÄ±n try ve catch sonrasÄ±nda kodu Ã§alÄ±ÅŸtÄ±rmanÄ±za izin verir.

<aside>
ğŸ’¡ JS Error nesnesi iki Ã¶zelliÄŸi olur; ad ve mesaj

</aside>

### Throw

```jsx
<!DOCTYPE html>
<html>
<body>

<p>Please input a number between 5 and 10:</p>

<input id="demo" type="text">
<button type="button" onclick="myFunction()">Test Input</button>
<p id="p01"></p>

<script>
function myFunction() {
  var message, x;
  message = document.getElementById("p01");
  message.innerHTML = "";
  x = document.getElementById("demo").value;
  try {
    if(x == "") throw "empty";
    if(isNaN(x)) throw "not a number";
    x = Number(x);
    if(x < 5) throw "too low";
    if(x > 10) throw "too high";
  }
  catch(err) {
    message.innerHTML = "Input is " + err;
  }
}
</script>

</body>
</html>

// EÄŸer ki sayÄ± 5,10 arasÄ±nda deÄŸilse : too low
```

### Finally

```jsx
function myFunction() {
  var message, x;
  message = document.getElementById("p01");
  message.innerHTML = "";
  x = document.getElementById("demo").value;
  try {
    if(x == "") throw "is empty";
    if(isNaN(x)) throw "is not a number";
    x = Number(x);
    if(x > 10) throw "is too high";
    if(x < 5) throw "is too low";
  }
  catch(err) {
    message.innerHTML = "Error: " + err + ".";
  }
  finally {
    document.getElementById("demo").value = "";
  }
}

// Hata varsa mesajÄ± yazdÄ±rÄ±r yoksa finally bloÄŸunu yazdÄ±rÄ±r.
```

[BazÄ± Hata Ã‡eÅŸitleri](JS%20Error%20e054bf63a3d049389df9bb5b6ff1edf4/Baz%C4%B1%20Hata%20C%CC%A7es%CC%A7itleri%208c3c9f88e3dc4783a899655123e3f99b.csv)

<aside>
â›” JS yeni sÃ¼rÃ¼mleri EvalError'u kullanmaz. Bunun yerine SyntaxError kullanÄ±lÄ±r.

</aside>